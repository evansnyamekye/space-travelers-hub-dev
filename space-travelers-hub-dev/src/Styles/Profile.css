.profile {
  padding: 20px;
  display: flex;
  justify-content: space-between;
  width: 50%;
}

.list {
  list-style-type: none;
  padding: 0;
  border-collapse: collapse;
}

.list li {
  border: 1px solid #ccc;
  width: 350px;
  padding: 10px;
}

.list li h3 {
  margin: 0;
}

h2 {
  margin-bottom: 20px;
}
