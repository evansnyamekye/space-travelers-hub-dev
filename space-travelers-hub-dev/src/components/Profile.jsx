import React from 'react';
import { useSelector } from 'react-redux';
import '../Styles/Profile.css';

const Profile = () => {
  const reservedRockets = useSelector(
    (state) => state.rockets.rockets.filter((rocket) => rocket.reserved),
  );
  const activeMissions = useSelector(
    (state) => state.missions.missions.filter((mission) => mission.mission_join),
  );

  return (
    <div className="profile">
      <div className="my-mission">
        <h2>My Missions</h2>
        <ul className="list">
          {activeMissions.map((mission) => (
            <li key={mission.mission_id}>
              <h3>{mission.mission_name}</h3>
            </li>
          ))}
        </ul>
      </div>
      <div className="my-rocket">
        <h2>My Rockets</h2>
        <ul className="list">
          {reservedRockets.map((rocket) => (
            <li key={rocket.id}>
              <h3>{rocket.rocket_name}</h3>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default Profile;
