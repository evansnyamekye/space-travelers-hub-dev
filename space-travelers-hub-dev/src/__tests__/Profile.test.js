import React from 'react';
import { useSelector } from 'react-redux';
import renderer from 'react-test-renderer';
import Profile from '../components/Profile';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Profile', () => {
  it('renders correctly', () => {
    useSelector.mockImplementation((selector) => selector({
      rockets: {
        rockets: [
          { id: 1, rocket_name: 'Rocket 1', reserved: true },
          { id: 2, rocket_name: 'Rocket 2', reserved: true },
        ],
      },
      missions: {
        missions: [
          { mission_id: 1, mission_name: 'Mission 1', mission_join: true },
          { mission_id: 2, mission_name: 'Mission 2', mission_join: true },
        ],
      },
    }));

    const component = renderer.create(<Profile />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
