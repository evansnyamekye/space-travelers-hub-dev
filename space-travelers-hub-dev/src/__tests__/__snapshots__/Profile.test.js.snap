// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<div
  className="profile"
>
  <div
    className="my-mission"
  >
    <h2>
      My Missions
    </h2>
    <ul
      className="list"
    >
      <li>
        <h3>
          Mission 1
        </h3>
      </li>
      <li>
        <h3>
          Mission 2
        </h3>
      </li>
    </ul>
  </div>
  <div
    className="my-rocket"
  >
    <h2>
      My Rockets
    </h2>
    <ul
      className="list"
    >
      <li>
        <h3>
          Rocket 1
        </h3>
      </li>
      <li>
        <h3>
          Rocket 2
        </h3>
      </li>
    </ul>
  </div>
</div>
`;
